index.js

```diff
@@ -1,10 +1,19 @@
 var self = require("sdk/self");
-var prefs = require("sdk/simple-prefs");
+var ss = require("sdk/simple-storage");
 
 const windowUtils = require("sdk/window/utils");
 var gBrowser = windowUtils.getMostRecentBrowserWindow().getBrowser();
 var gWindow = windowUtils.getMostRecentBrowserWindow();
-var gMode = prefs.prefs.mode;
+
+// load and validate settings
+var gMode = ss.storage.mode;
+if (gMode !== "threadHangs" && gMode !== "eventLoopLags") {
+	gMode = "threadHangs";
```

Comment:
Nit - let's stick with two space indentation. Goes for the rest of your changes as well.